x-pack/plugins/maps/public/components/geo_index_pattern_select.tsx

```diff
@@ -106,7 +106,8 @@ export function GeoIndexPatternSelect(props: Props) {
   }
 
   const IndexPatternSelect = getIndexPatternSelectComponent();
-  const error = !hasGeoFields
+  const isDataViewInvalid = !props.dataView ? false : !hasGeoFields;
```

Comment:
Fix for this

<img width="200" alt="Screen Shot 2023-06-22 at 3 40 12 PM" src="https://github.com/elastic/kibana/assets/373691/ea656801-9c53-4779-a66c-d345f2759b66">